### DISCUSSION TITLE
ROS2 support

### DISCUSSION THREAD

**Discussion by christianlandgraf** [sam-xl/scancontrol#42] [General]
Hi,

out of interest: Are there any plans to provide ROS2 support for the scancontrol sensors? :)

**Comment by dave992** [comment]
At the moment we do not have any plans to port the driver to ROS 2 in the near future. 

It should however not be very hard to do, as this driver is in essence just a wrapper around their SDK.

**Comment by christianlandgraf** [comment]
I started to migrate the driver to ROS2: https://github.com/christianlandgraf/scancontrol/tree/ros2-devel
It managed to receive data from a scanCONTROL 2900-100 device.
I didn't touch the Nodelet part and some of the tests, yet. Since the `pcl_ros` package is not fully available in ROS2, I had to add another Point Cloud format conversion. Although it still published the data with 100 hz in my first tests, I am not sure if this may lead to perfomance issues in more comprehensive tests.

Since I am completely new to ROS2, there also might exists further improvements...

**Comment by christianlandgraf** [comment]
If you are interested, I could further finish the work and merge it into a "ros2" branch in this repository ( I guess it shouldn't be merged into the master branch)

**Comment by dave992** [comment]
We would be open to that for sure, I can make a new branch for this and if your changes are ready you can submit an MR that we can review. Thanks!

**Comment by nikhil-sethi** [comment]
Check out branch `ros2-devel`. The port of ROS2 is being actively worked on with new features
